internal/tracer.go

```diff
@@ -28,25 +28,34 @@ import (
 	"context"
 
 	"github.com/opentracing/opentracing-go"
+	commonpb "go.temporal.io/temporal-proto/common"
 	"go.uber.org/zap"
 )
 
+// Must impement opentracing.TextMapReader
```

Comment:
This is not obvious.

Reply:
May be add real check that would break compilation if not implemented?